README.md

```diff
@@ -1 +1,107 @@
-# gha-ci
\ No newline at end of file
+# GitHub Actions - CI
+
+CI for Silverstripe modules
+
+Uses feature detection based on files in the root folder such as phpunit.xml.dist to build and run a dynamic matrix of jobs.
+
+It's highly recommended that you use a tagged version (e.g. v1) to ensure stability of your builds. If you have a relatively simple build that you have no intention of ever making more complex e.g. only phpunit tests using phpunit.xml.dist, then this is probably all you need for long term use.
+
+Note: Unlike other silverstripe/gha-* repositories, this one is a [reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows) rather than an [action](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action). A reusable workflow is required to create a `matrix`.
+
+### Usage
+
+Create the following file in your module, and substitute the tagged version for the most recent tag prefixed with a `v` e.g. `@v1`
+
+**.github/workflows/ci.yml**
+```yml
+name: CI
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  ci:
+    uses: silverstripe/gha-ci/.github/workflows/ci.yml@main
+```
+
+Set config specific to your needs via "inputs" defined under the `with:` key. For instance, to disable PHP linting because your module does not does not yet have a `phpcs.xml.dist` file
```

Comment:
```suggestion
Set config specific to your needs via "inputs" defined under the `with:` key. For instance, to disable PHP linting because your module does not yet have a `phpcs.xml.dist` file
```
Sorry, should have noticed this last time through

Reply:
Updated